File: Leetcode/FindSmallestLetterGreaterThanTarget.rs
// https://leetcode.com/problems/find-smallest-letter-greater-than-target

struct Solution {}

impl Solution {
    pub fn next_greatest_letter(letters: Vec<char>, target: char) -> char {
        *letters
            .iter()
            .find(|c| c > &&target)
            .unwrap_or(letters.first().unwrap())
    }
}

File: Leetcode/FindSmallestLetterGreaterThanTarget.c
// https://leetcode.com/problems/find-smallest-letter-greater-than-target

char nextGreatestLetter(char* letters, int lettersSize, char target) {
    int l = -1;
    int r = lettersSize;

    while (r - l > 1) {
        int m = (r + l) >> 1;

        if (letters[m] <= target) {
            l = m;
        } else {
            r = m;
        }
    }

    return letters[r % lettersSize];
}